pages/gateway/GatewayHandler.inc.php

```diff
@@ -10,12 +10,36 @@
  * @class GatewayHandler
  * @ingroup pages_gateway
  *
- * @brief Handle external gateway requests. 
+ * @brief Handle external gateway requests.
  */
 
 import('classes.handler.Handler');
 
 class GatewayHandler extends Handler {
+
+	/**
+	 * Constructor
+	 *
+	 * @param $request PKPRequest
+	 */
+	function __construct($request) {
+		parent::__construct();
+
+		$op = $request->getRouter()->getRequestedOp($request);
+		if ($op == 'plugin') {
+			$pluginName = array_slice($request->getRouter()->getRequestedArgs($request), 0, 1)[0];
```

Comment:
Isn't this just equivalent to `array_shift($request->getRouter()->getRequestedArgs($request)`?

Reply:
Yes.